priv/stdlib/List.aes

```diff
@@ -37,14 +37,18 @@ namespace List =
     []   => reverse(acc)
     h::t => find_indices_(p, t, n+1, if(p(h)) n::acc else acc)
 
-  function nth(n : int, l : list('a)) : option('a) = switch(l)
-    []   => None
-    h::t => if(n == 0) Some(h) else nth(n-1, t)
+  function nth(n : int, l : list('a)) : option('a) =
+    if(n < 0) None
+    else switch(l)
+      []   => None
+      h::t => if(n == 0) Some(h) else nth(n-1, t)
```

Comment:
What's the motivation for this change? The semantics hasn't changed and the function is slower for legitimate use cases.

Reply:
More meaningful abort message for example. However, the check `n < 0` could be after list traversal

Reply:
Rollbacked